tests/beaker_tests/snmp_notification_receiver/snmp_notification_receiver_provider_defaults.rb

```diff
@@ -0,0 +1,188 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###############################################################################
+# TestCase Name:
+# -------------
+# SnmpNotificationReceiver-Provider-Defaults.rb
+#
+# TestCase Prerequisites:
+# -----------------------
+# This is a snmp_notification_receiver resource testcase for Puppet Agent on Nexus devices.
+# The test case assumes the following prerequisites are already satisfied:
+# A. Populating the HOSTS configuration file with the agent and master
+# information.
+# B. Enabling SSH connection prerequisites on the N9K switch based Agent.
+# C. Starting of Puppet master server on master.
+# D. Sending to and signing of Puppet agent certificate request on master.
+#
+# TestCase:
+# ---------
+# This is a snmp_notification_receiver resource test that tests for default value for
+# 'ensure' attribute of a snmp_notification_receiver resource.
+#
+# The testcode checks for exit_codes from Puppet Agent, Vegas shell and
+# Bash shell command executions. For Vegas shell and Bash shell command
+# string executions, this is the exit_code convention:
+# 0 - successful command execution, > 0 - failed command execution.
+# For Puppet Agent command string executions, this is the exit_code convention:
+# 0 - no changes have occurred, 1 - errors have occurred,
+# 2 - changes have occurred, 4 - failures have occurred and
+# 6 - changes and failures have occurred.
+# 0 is the default exit_code checked in Beaker::DSL::Helpers::on() method.
+# The testcode also uses RegExp pattern matching on stdout or output IO
+# instance attributes of Result object from on() method invocation.
+#
+###############################################################################
+
+# Require UtilityLib.rb and SnmpGroupLib.rb paths.
+require File.expand_path('../../lib/utilitylib.rb', __FILE__)
+require File.expand_path('../snmp_notification_receiverlib.rb', __FILE__)
+
+result = 'PASS'
+testheader = 'snmp_notification_receiver Resource :: All Attributes Defaults'
+
+# @test_name [TestCase] Executes defaults testcase for snmp_notification_receiver Resource.
+test_name "TestCase :: #{testheader}" do
+  # @step [Step] Sets up switch for provider test.
+  step 'TestStep :: Setup switch for provider' do
+    # Define PUPPETMASTER_MANIFESTPATH constant using puppet config cmd.
+    UtilityLib.set_manifest_path(master, self)
+
+    # Ensure that resource is removed beforehand
+    on(master, SnmpNotificationReceiverLib.create_snmp_notification_receiver_manifest_absent)
```

Comment:
Instead of this cleanup logic, we now have a new method that will remove all instances of a particular resource.  Let's use that instead.

Here is an example of how it can be called.

https://github.com/cisco/cisco-network-puppet-module/blob/develop/tests/beaker_tests/cisco_vxlan_vtep/test_vxlan_vtep.rb#L215